compiler/plugins/input/StableHLO/stablehlo-iree/Conversion/StableHLOCustomCalls.cpp

```diff
@@ -221,6 +310,242 @@ struct HouseholderReflectorRewriter final
+                           Value div =
+                               bb.create<arith::DivFOp>(loc, args[1], sqrt);
+                           bb.create<linalg::YieldOp>(loc, div);
+                         })
+                        .getResult(0);
+
+    // We initialize the tensor with all zeros.
+    Value zeroF =
+        b.create<arith::ConstantOp>(b.getZeroAttr(resultTy.getElementType()));
+    Value empty = b.create<tensor::SplatOp>(resultTy, zeroF);
+
+    Value insert = insertSlice(empty, generic, {}, {}, b);
+
+    // Iterate row-by-row in the input matrix and upate the estimated
+    // cholesky decomposition.
+    Value one = b.create<arith::ConstantIndexOp>(1);
+    Value N = b.create<arith::ConstantIndexOp>(operandTy.getShape().back());
+    Value scf =
+        b.create<scf::ForOp>(
+             one, N, one, ValueRange{insert},
+             [&](OpBuilder &bb, Location loc, Value col, ValueRange args) {
+               ImplicitLocOpBuilder b(loc, bb);
+               // vecvec = dot(L[i, :], L[i, :]):
+               Value outSlice =
+                   extractSlice(args[0], {col}, {1}, {rank - 2}, b);
+               outSlice = collapseBackDims(outSlice, 1, b);
+               Value vecvec = makeDot(outSlice, outSlice, rank - 2, 1, b);
+
+               // matvec = dot(L[:, :], L[i, :]):
+               Value inSlice = extractSlice(operand, {col}, {1}, {rank - 2}, b);
+               inSlice = collapseBackDims(inSlice, 1, b);
+               Value matvec = makeDot(args[0], outSlice, rank - 2, 1, b);
+
+               // diag = A[i, i]
+               Value diag = extractSlice(operand, {col, col}, {1, 1},
+                                         {rank - 2, rank - 1}, b);
+               diag = collapseBackDims(diag, 2, b);
+
+               AffineMap scalarMap = getAffineMap({rank - 2}, rank - 1, b);
+               AffineMap outMap = b.getMultiDimIdentityMap(rank - 1);
+               SmallVector<AffineMap> affineMaps = {
+                   outMap, outMap, outMap, scalarMap, scalarMap, outMap};
+               SmallVector<utils::IteratorType> iterTypes(
+                   rank - 1, utils::IteratorType::parallel);
+
+               ShapedType colTy = RankedTensorType::get(
+                   cast<ShapedType>(resultTy).getShape().drop_back(),
+                   resultTy.getElementType());
+               Value empty = b.create<tensor::EmptyOp>(colTy.getShape(),
+                                                       colTy.getElementType());
+
+               // Compute the new column values:
+               //   C_i = sqrt(diag - vecvec)
+               //   C[j] = j <= i ? matvec[j] - L[i, j] : 0
+               Value generic =
+                   b.create<linalg::GenericOp>(
+                        colTy,
+                        ValueRange{outSlice, matvec, inSlice, diag, vecvec},
+                        empty, affineMaps, iterTypes,
+                        [&](OpBuilder &bb, Location loc, ValueRange args) {
```

Comment:
In class function definition with 2 nested lamdas eats a rather lot of whitespace ... If you defined the function post the class you would at least get 2 hspaces back and perhaps more fit on line. And then consider making the outer most lambda a variable (I think you want to keep it all together rather than having multiple to see the flow more easily, but the outermost lamda being inside the b.create is costing about 8 hspace whitespace)